plugins/processors/enum/enum.go

```diff
@@ -37,17 +39,38 @@ func (mapper *EnumMapper) Init() error {
 	mapper.FieldFilters = make(map[string]filter.Filter)
 	mapper.TagFilters = make(map[string]filter.Filter)
 	for _, mapping := range mapper.Mappings {
+		if mapping.Field != "" && len(mapping.Fields) != 0 {
+			return fmt.Errorf("use field or fields, but not both")
+		}
 		if mapping.Field != "" {
 			fieldFilter, err := filter.NewIncludeExcludeFilter([]string{mapping.Field}, nil)
 			if err != nil {
-				return fmt.Errorf("failed to create new field filter: %w", err)
+				return fmt.Errorf("failed to create new field filter from field: %w", err)
 			}
 			mapper.FieldFilters[mapping.Field] = fieldFilter
 		}
+		if len(mapping.Fields) != 0 {
+			fieldFilter, err := filter.NewIncludeExcludeFilter(mapping.Fields, nil)
+			if err != nil {
+				return fmt.Errorf("failed to create new field filter from fields: %w", err)
+			}
+			mapper.FieldFilters[mapping.Field] = fieldFilter
```

Comment:
I don't understand why you use a map for the filter... The key for the map is either empty (if no field is defined) or has the field name, i.e. the map will only contain a single entry. I think you should store the filter directly instead of using a map.